pyrokinetics/gene.py

```diff
@@ -0,0 +1,379 @@
+
+        # Kinetic data
+        local_species = pyro.local_species
+        gene_input['box']['nspec'] = local_species.nspec
+
+        pyro_gene_species = self.pyro_to_gene_species()
+
+        for iSp, name in enumerate(local_species.names):
+
+            species_key = 'species'  
+
+            if name == 'electron':
+                gene_input[species_key][iSp]['name'] = 'electron'
+            else:
+                try:
+                    gene_input[species_key][iSp]['name'] = 'ion'
+                except KeyError:
+                    gene_input[species_key] = copy.copy(gene_input['species_1'])
+                    gene_input[species_key]['name'] = 'ion'
+            
+            for key, val in pyro_gene_species.items():
+                gene_input[species_key][iSp][val] = local_species[name][key]
+
+        # If species are defined calculate beta
+        if local_species.nref is not None:
+
+            pref = local_species.nref * local_species.tref * electron_charge
+
+            beta = pref/bref**2 * 8 * pi * 1e-7
+
+        # Calculate from reference  at centre of flux surface
+        else:
+            if pyro.local_geometry_type == 'Miller':
+                if miller.B0 is not None:
+                    beta = 1 / miller.B0** 2 * (miller.Rgeo / miller.Rmaj) ** 2
+                else:
+                    beta = 0.0
+            else:
+                raise NotImplementedError
+
+        gene_input['general']['beta'] = beta
+
+        # Numerics
+        numerics = pyro.numerics
+
+        if numerics['nonlinear']:
+            
+            gene_input['general']['nonlinear']= True
+            gene_input['box']['nky0'] = numerics['nky']
+            gene_input['box']['nkx'] = numerics['nkx']
+
+        else:
+            gene_input['general']['nonlinear'] = False
+
+        # Currently forces NL sims to have nperiod = 1
+        gene_nml = f90nml.Namelist(gene_input)
+        gene_nml.float_format = pyro.float_format
+        gene_nml.write(path_to_file, force=True)
+
+    
+    def load_local_geometry(self, pyro, gene):
+        """
+        Loads local geometry 
+        """
+
+        if pyro.local_geometry_type == 'Miller':
+            self.load_miller(pyro,gene)
+
+    def load_miller(self, pyro, gene):
+        """
+        Load Miller object from GENE file
+        """
+
+        # Set some defaults here
+        gene['geometry']['magn_geometry'] = 'miller'
+
+        pyro_gene_miller = self.pyro_to_gene_miller()
+
+        miller = pyro.local_geometry
+
+        for key, val in pyro_gene_miller.items():
+            miller[key] = gene[val[0]][val[1]]
+
+        miller.kappri = miller.s_kappa*miller.kappa / miller.rho
+        miller.tri = np.arcsin(miller.delta)
+
+        # Get beta normalised to R_major(in case R_geo != R_major)
+        beta = gene['general']['beta'] * (miller.Rmaj/miller.Rgeo)**2
+
+        # Assume pref*8pi*1e-7 = 1.0
+        if beta != 0.0:
+            miller.B0 = np.sqrt(1.0/beta**0.5)
```

Comment:
This was a mistake on my front, double square root

```suggestion
            miller.B0 = np.sqrt(1.0/beta)
```